docs/datamodel/links.rst

```diff
@@ -134,6 +139,36 @@ Parameters:
 :eschema:synopsis:`<constraint-declarations>`
     :ref:`Constraint <ref_datamodel_constraints>` declarations.
 
+.. _ref_datamodel_links_on_delete:
+
+On target delete options cover the situation when the target object of
+a link is deleted without explicitly updating the link:
+
+:eschema:synopsis:`restrict`
+    Prohibit deleting the link target as long as the parent object exists.
+    This is the default behavior.
+
+:eschema:synopsis:`delete source`
+    Delete the parent object if any link target is deleted. This may
+    not be desirable behavior for ``multi`` links since the parent
+    object will be deleted if *just* one of the link targets is
+    deleted.
+
+:eschema:synopsis:`set empty`
+    Set the link to be empty when the target object is deleted. This
+    may not be desirable behavior for ``multi`` links since the link
+    will be set to an empty set if *just* one of the link targets is
```

Comment:
not sure what "just one of the link targets deleted" means here

Reply:
Reading this I’m starting to think that `set empty` is a misnomer. What we actually *want* here is to simply delete the corresponding link and nothing else. So, yeah, the link set becomes *empty* if the last target was removed, but that is a side effect, not an explicit action itself. We should rename the action to `accept` or `delete link` or something along these lines.

Similarly, I think `delete source` on a multilink doesn’t strike me like a particularly useful thing. Perhaps we should ban this for now.

Reply:
`delete source` for multilinks is perhaps more unusual, but it's perfectly reasonable for situations where the multilink belongs to some grouping where the integrity of the group is paramount: like a competition team (can't have incomplete teams). 

I'll drop the `set default` as it seems to be broken right now.

I'll rename `set empty` to `accept`.

Reply:
Maybe `allow` is better than `accept`? Our default mode is "restrict" so "accept/reject" vs. "allow/restrict".

Reply:
I've updated the PR. It's no longer just documentation, but also has parser and test changes.